{
  "Software_Repo_MappingId": null,
  "Software_RepoId": "d9fcf5c5-e1ff-4feb-8eb3-61341c4ab05c",
  "Software_Name": "Node.js",
  "CVEID": "CVE-2021-4044",
  "Summary": "Internally libssl in OpenSSL calls X509_verify_cert() on the client side to verify a certificate supplied by a server. That function may return a negative return value to indicate an internal error (for example out of memory). Such a negative return value is mishandled by OpenSSL and will cause an IO function (such as SSL_connect() or SSL_do_handshake()) to not indicate success and a subsequent call to SSL_get_error() to return the value SSL_ERROR_WANT_RETRY_VERIFY. This return value is only supposed to be returned by OpenSSL if the application has previously called SSL_CTX_set_cert_verify_callback(). Since most applications do not do this the SSL_ERROR_WANT_RETRY_VERIFY return value from SSL_get_error() will be totally unexpected and applications may not behave correctly as a result. The exact behaviour will depend on the application but it could result in crashes, infinite loops or other similar incorrect responses. This issue is made more serious in combination with a separate bug in OpenSSL 3.0 that will cause X509_verify_cert() to indicate an internal error when processing a certificate chain. This will occur where a certificate does not include the Subject Alternative Name extension but where a Certificate Authority has enforced name constraints. This issue can occur even with valid chains. By combining the two issues an attacker could induce incorrect, application dependent behaviour. Fixed in OpenSSL 3.0.1 (Affected 3.0.0).",
  "LastUpdate": 0,
  "CVEE": {
    "CVEEId": "CVE-2021-4044",
    "CVSS_V2_baseScore": 5.0,
    "CVSS_V2_Data": "{\"version\":\"2.0\",\"baseScore\":5.0,\"accessVector\":\"NETWORK\",\"vectorString\":\"AV:N/AC:L/Au:N/C:N/I:N/A:P\",\"authentication\":\"NONE\",\"integrityImpact\":\"NONE\",\"accessComplexity\":\"LOW\",\"availabilityImpact\":\"PARTIAL\",\"confidentialityImpact\":\"NONE\",\"scope\":null,\"attackVector\":null,\"baseSeverity\":null,\"userInteraction\":null,\"attackComplexity\":null,\"privilegesRequired\":null}",
    "CVSS_V3_baseScore": 7.5,
    "CVSS_V3_Data": "{\"version\":\"3.1\",\"baseScore\":7.5,\"accessVector\":null,\"vectorString\":\"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H\",\"authentication\":null,\"integrityImpact\":\"NONE\",\"accessComplexity\":null,\"availabilityImpact\":\"HIGH\",\"confidentialityImpact\":\"NONE\",\"scope\":\"UNCHANGED\",\"attackVector\":\"NETWORK\",\"baseSeverity\":\"HIGH\",\"userInteraction\":\"NONE\",\"attackComplexity\":\"LOW\",\"privilegesRequired\":\"NONE\"}",
    "PublishedDate": "2021-12-14T19:15:07.807",
    "ModifyDate": "2024-11-21T06:36:47.243",
    "VulnStatus": "Modified",
    "JsonData": "{\"id\": \"CVE-2021-4044\", \"summary\": \"Internally libssl in OpenSSL calls X509_verify_cert() on the client side to verify a certificate supplied by a server. That function may return a negative return value to indicate an internal error (for example out of memory). Such a negative return value is mishandled by OpenSSL and will cause an IO function (such as SSL_connect() or SSL_do_handshake()) to not indicate success and a subsequent call to SSL_get_error() to return the value SSL_ERROR_WANT_RETRY_VERIFY. This return value is only supposed to be returned by OpenSSL if the application has previously called SSL_CTX_set_cert_verify_callback(). Since most applications do not do this the SSL_ERROR_WANT_RETRY_VERIFY return value from SSL_get_error() will be totally unexpected and applications may not behave correctly as a result. The exact behaviour will depend on the application but it could result in crashes, infinite loops or other similar incorrect responses. This issue is made more serious in combination with a separate bug in OpenSSL 3.0 that will cause X509_verify_cert() to indicate an internal error when processing a certificate chain. This will occur where a certificate does not include the Subject Alternative Name extension but where a Certificate Authority has enforced name constraints. This issue can occur even with valid chains. By combining the two issues an attacker could induce incorrect, application dependent behaviour. Fixed in OpenSSL 3.0.1 (Affected 3.0.0).\", \"created_at\": \"2021-12-14T19:15:07Z\", \"updated_at\": \"2024-11-21T06:36:47Z\", \"cvss\": {\"v2\": 5.0, \"v3\": 7.5}, \"vendors\": {\"netapp\": [\"h500e_firmware\", \"h700s\", \"h700e\", \"h700s_firmware\", \"500f_firmware\", \"ontap_select_deploy_administration_utility\", \"h500s_firmware\", \"h410s\", \"h410s_firmware\", \"snapcenter\", \"500f\", \"h300s\", \"h700e_firmware\", \"h410c_firmware\", \"a250_firmware\", \"cloud_backup\", \"h500s\", \"h300s_firmware\", \"h500e\", \"h300e\", \"h410c\", \"a250\", \"e-series_performance_analyzer\", \"h300e_firmware\"], \"nodejs\": [\"node.js\"], \"openssl\": [\"openssl\"]}, \"cwes\": [\"CWE-835\"], \"raw_nvd_data\": {\"id\": \"CVE-2021-4044\", \"metrics\": {\"cvssMetricV2\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"cvssData\": {\"version\": \"2.0\", \"baseScore\": 5.0, \"accessVector\": \"NETWORK\", \"vectorString\": \"AV:N/AC:L/Au:N/C:N/I:N/A:P\", \"authentication\": \"NONE\", \"integrityImpact\": \"NONE\", \"accessComplexity\": \"LOW\", \"availabilityImpact\": \"PARTIAL\", \"confidentialityImpact\": \"NONE\"}, \"acInsufInfo\": false, \"impactScore\": 2.9, \"baseSeverity\": \"MEDIUM\", \"obtainAllPrivilege\": false, \"exploitabilityScore\": 10.0, \"obtainUserPrivilege\": false, \"obtainOtherPrivilege\": false, \"userInteractionRequired\": false}], \"cvssMetricV31\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"cvssData\": {\"scope\": \"UNCHANGED\", \"version\": \"3.1\", \"baseScore\": 7.5, \"attackVector\": \"NETWORK\", \"baseSeverity\": \"HIGH\", \"vectorString\": \"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H\", \"integrityImpact\": \"NONE\", \"userInteraction\": \"NONE\", \"attackComplexity\": \"LOW\", \"availabilityImpact\": \"HIGH\", \"privilegesRequired\": \"NONE\", \"confidentialityImpact\": \"NONE\"}, \"impactScore\": 3.6, \"exploitabilityScore\": 3.9}]}, \"published\": \"2021-12-14T19:15:07.807\", \"references\": [{\"url\": \"https://git.openssl.org/gitweb/?p=openssl.git%3Ba=commitdiff%3Bh=758754966791c537ea95241438454aa86f91f256\", \"source\": \"openssl-security@openssl.org\"}, {\"url\": \"https://security.netapp.com/advisory/ntap-20211229-0003/\", \"tags\": [\"Vendor Advisory\"], \"source\": \"openssl-security@openssl.org\"}, {\"url\": \"https://www.openssl.org/news/secadv/20211214.txt\", \"tags\": [\"Vendor Advisory\"], \"source\": \"openssl-security@openssl.org\"}, {\"url\": \"https://git.openssl.org/gitweb/?p=openssl.git%3Ba=commitdiff%3Bh=758754966791c537ea95241438454aa86f91f256\", \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://security.netapp.com/advisory/ntap-20211229-0003/\", \"tags\": [\"Vendor Advisory\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}, {\"url\": \"https://www.openssl.org/news/secadv/20211214.txt\", \"tags\": [\"Vendor Advisory\"], \"source\": \"af854a3a-2127-422b-91ae-364da2661108\"}], \"vulnStatus\": \"Modified\", \"weaknesses\": [{\"type\": \"Primary\", \"source\": \"nvd@nist.gov\", \"description\": [{\"lang\": \"en\", \"value\": \"CWE-835\"}]}], \"descriptions\": [{\"lang\": \"en\", \"value\": \"Internally libssl in OpenSSL calls X509_verify_cert() on the client side to verify a certificate supplied by a server. That function may return a negative return value to indicate an internal error (for example out of memory). Such a negative return value is mishandled by OpenSSL and will cause an IO function (such as SSL_connect() or SSL_do_handshake()) to not indicate success and a subsequent call to SSL_get_error() to return the value SSL_ERROR_WANT_RETRY_VERIFY. This return value is only supposed to be returned by OpenSSL if the application has previously called SSL_CTX_set_cert_verify_callback(). Since most applications do not do this the SSL_ERROR_WANT_RETRY_VERIFY return value from SSL_get_error() will be totally unexpected and applications may not behave correctly as a result. The exact behaviour will depend on the application but it could result in crashes, infinite loops or other similar incorrect responses. This issue is made more serious in combination with a separate bug in OpenSSL 3.0 that will cause X509_verify_cert() to indicate an internal error when processing a certificate chain. This will occur where a certificate does not include the Subject Alternative Name extension but where a Certificate Authority has enforced name constraints. This issue can occur even with valid chains. By combining the two issues an attacker could induce incorrect, application dependent behaviour. Fixed in OpenSSL 3.0.1 (Affected 3.0.0).\"}, {\"lang\": \"es\", \"value\": \"Internamente libssl en OpenSSL llama a X509_verify_cert() en el lado del cliente para verificar un certificado suministrado por un servidor. Esta funci\\u00f3n puede devolver un valor negativo para indicar un error interno (por ejemplo, falta de memoria). Tal valor de retorno negativo es mal manejado por OpenSSL y causar\\u00e1 que una funci\\u00f3n IO (como SSL_connect() o SSL_do_handshake()) no indique el \\u00e9xito y una llamada posterior a SSL_get_error() devuelva el valor SSL_ERROR_WANT_RETRY_VERIFY. Este valor de retorno s\\u00f3lo debe ser devuelto por OpenSSL si la aplicaci\\u00f3n ha llamado previamente a SSL_CTX_set_cert_verify_callback(). Como la mayor\\u00eda de las aplicaciones no hacen esto, el valor de retorno SSL_ERROR_WANT_RETRY_VERIFY de SSL_get_error() ser\\u00e1 totalmente inesperado y las aplicaciones pueden no comportarse correctamente como resultado. El comportamiento exacto depender\\u00e1 de la aplicaci\\u00f3n, pero podr\\u00eda resultar en bloqueos, bucles infinitos u otras respuestas incorrectas similares. Este problema se agrava en combinaci\\u00f3n con otro fallo en OpenSSL versi\\u00f3n 3.0 que har\\u00e1 que X509_verify_cert() indique un error interno cuando procesa una cadena de certificados. Esto ocurrir\\u00e1 cuando un certificado no incluya la extensi\\u00f3n de nombre alternativo del sujeto pero cuando una autoridad de certificaci\\u00f3n haya aplicado restricciones de nombre. Este problema puede producirse incluso con cadenas v\\u00e1lidas. Combinando los dos problemas, un atacante podr\\u00eda inducir un comportamiento incorrecto y dependiente de la aplicaci\\u00f3n. Corregido en OpenSSL versi\\u00f3n 3.0.1 (Afectado 3.0.0)\"}], \"lastModified\": \"2024-11-21T06:36:47.243\", \"configurations\": [{\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:a:openssl:openssl:*:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"D53A6288-46D2-452F-95DA-ADA3A55544E5\", \"versionEndExcluding\": \"1.0.2\"}, {\"criteria\": \"cpe:2.3:a:openssl:openssl:1.1.0:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"73104834-5810-48DD-9B97-549D223853F1\"}, {\"criteria\": \"cpe:2.3:a:openssl:openssl:3.0.0:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"5D1E839A-4780-412E-9F02-DD3029A0B8EF\"}], \"operator\": \"OR\"}]}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:a:netapp:cloud_backup:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"5C2089EE-5D7F-47EC-8EA5-0F69790564C4\"}, {\"criteria\": \"cpe:2.3:a:netapp:e-series_performance_analyzer:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"24B8DB06-590A-4008-B0AB-FCD1401C77C6\"}, {\"criteria\": \"cpe:2.3:a:netapp:ontap_select_deploy_administration_utility:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"E7CF3019-975D-40BB-A8A4-894E62BD3797\"}, {\"criteria\": \"cpe:2.3:a:netapp:snapcenter:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"BDFB1169-41A0-4A86-8E4F-FDA9730B1E94\"}], \"operator\": \"OR\"}]}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:a250_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"1236B66D-EB11-4324-929F-E2B86683C3C7\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:a250:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"281DFC67-46BB-4FC2-BE03-3C65C9311F65\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:500f_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"ECF32BB1-9A58-4821-AE49-5D5C8200631F\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:500f:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"F21DE67F-CDFD-4D36-9967-633CD0240C6F\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:h500s_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"7FFF7106-ED78-49BA-9EC5-B889E3685D53\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:h500s:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"E63D8B0F-006E-4801-BF9D-1C001BBFB4F9\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:h700s_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"56409CEC-5A1E-4450-AA42-641E459CC2AF\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:h700s:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"B06F4839-D16A-4A61-9BB5-55B13F41E47F\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:h300e_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"108A2215-50FB-4074-94CF-C130FA14566D\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:h300e:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"7AFC73CE-ABB9-42D3-9A71-3F5BC5381E0E\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:h500e_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"32F0B6C0-F930-480D-962B-3F4EFDCC13C7\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:h500e:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"803BC414-B250-4E3A-A478-A3881340D6B8\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:h700e_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"0FEB3337-BFDE-462A-908B-176F92053CEC\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:h700e:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"736AEAE9-782B-4F71-9893-DED53367E102\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:h410s_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"D0B4AD8A-F172-4558-AEC6-FF424BA2D912\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:h410s:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"8497A4C9-8474-4A62-8331-3FE862ED4098\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:h410c_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"234DEFE0-5CE5-4B0A-96B8-5D227CB8ED31\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:h410c:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"CDDF61B7-EC5C-467C-B710-B89F502CD04F\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:o:netapp:h300s_firmware:-:*:*:*:*:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"6770B6C3-732E-4E22-BF1C-2D2FD610061C\"}], \"operator\": \"OR\"}, {\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:h:netapp:h300s:-:*:*:*:*:*:*:*\", \"vulnerable\": false, \"matchCriteriaId\": \"9F9C8C20-42EB-4AB5-BD97-212DEB070C43\"}], \"operator\": \"OR\"}], \"operator\": \"AND\"}, {\"nodes\": [{\"negate\": false, \"cpeMatch\": [{\"criteria\": \"cpe:2.3:a:nodejs:node.js:*:*:*:*:-:*:*:*\", \"vulnerable\": true, \"matchCriteriaId\": \"EF176509-5AA8-4644-84E5-051964AECCE5\", \"versionEndExcluding\": \"17.3.0\", \"versionStartIncluding\": \"17.0.0\"}], \"operator\": \"OR\"}]}], \"sourceIdentifier\": \"openssl-security@openssl.org\"}}\n",
    "LastUpdate": 0,
    "VendorL": [
      "netapp",
      "nodejs",
      "openssl"
    ],
    "ProductL": [
      "h500e_firmware",
      "h700s",
      "h700e",
      "h700s_firmware",
      "500f_firmware",
      "ontap_select_deploy_administration_utility",
      "h500s_firmware",
      "h410s",
      "h410s_firmware",
      "snapcenter",
      "500f",
      "h300s",
      "h700e_firmware",
      "h410c_firmware",
      "a250_firmware",
      "cloud_backup",
      "h500s",
      "h300s_firmware",
      "h500e",
      "h300e",
      "h410c",
      "a250",
      "e-series_performance_analyzer",
      "h300e_firmware",
      "node.js",
      "openssl"
    ],
    "Software_Repo_Mappings": []
  },
  "Software_Repo_Mapping_Conditions": [
    {
      "Software_Repo_Mapping_ConditionId": 0,
      "Software_Repo_MappingId": null,
      "Criteria": "cpe:2.3:a:nodejs:node.js:*:*:*:*:-:*:*:*",
      "VersionEndExcluding": "17.3.0",
      "VersionEndIncluding": null,
      "VersionStartExcluding": null,
      "VersionStartIncluding": "17.0.0",
      "Vulnerable": true,
      "Software_Repo_Mapping": null
    }
  ]
}